src/App/Pages/Accounts/TwoFactorPageViewModel.cs

```diff
@@ -356,18 +348,34 @@ public async Task SubmitAsync(bool showLoading = true)
                         }
                         else
                         {
+                            _syncService.FullSyncAsync(true).FireAndForget();
                             await TwoFactorAuthSuccessAsync();
                         }
                     }
                     else
                     {
                         StartDeviceApprovalOptionsAction?.Invoke();
                     }
+                    return;
                 }
-                else
+
+                // In the standard, non TDE case, a user must set password if they don't
+                // have one and they aren't using key connector.
+                // Note: TDE & Key connector are mutually exclusive org config options.
+                if (_authingWithSso && (result.ResetMasterPassword || (decryptOptions?.RequireSetPassword ?? false)))
```

Comment:
Do we need to check `_authingWithSso` here? I don't think it matters if you're using sso to login or not, we'll still need to set your password if these are true.

Does the server have the logic to only send `ResetMasterPassword` or `decryptOptions.RequireSetPassword` if the user isn't using Key Connector? I just want to make sure we don't need that logic here.